Capri.Services/Proposals/ProposalUpdater.cs

```diff
@@ -23,37 +23,43 @@ public ProposalUpdater(ISqlDbContext context, IMapper mapper, IUserGetter userGe
             _userGetter = userGetter;
             _submittedProposalGetter = submittedProposalGetter;
         }
-        public async Task<IServiceResult<Proposal>> Update(Guid id, ProposalUpdate proposalUpdate)
+        public async Task<IServiceResult<ProposalView>> Update(
+            Guid id, 
+            ProposalRegistration inputData)
         {
-            var currentUser = await _userGetter.GetCurrentUser();
-            var currentPromoter = await _context.Promoters.FirstOrDefaultAsync(p => p.UserId == currentUser.Id);
-
-            if (proposalUpdate.Level == StudyLevel.Master)
+            var result = await _userGetter.GetCurrentUser();
+            if(!result.Successful())
             {
-                if (_submittedProposalGetter.GetMasterProposalNumber(currentPromoter.Id).Result.Body() >=
-                    currentPromoter.ExpectedNumberOfMasterProposals)
-                    return ServiceResult<Proposal>.Error("You do not have permissions to create master proposal.");
+                var errors = result.GetAggregatedErrors();
+                return ServiceResult<ProposalView>.Error(errors);
             }
-            else
+            
+            var currentUser = result.Body();
+            var promoter = 
+                await _context
+                .Promoters
+                .Include(p => p.Proposals)
+                .FirstOrDefaultAsync(p => p.UserId == currentUser.Id);
```

Comment:
Same as above, can't you specify also here second condition?

Reply:
done